src/components/AssetHeader/AssetActions.tsx

```diff
@@ -29,6 +31,7 @@ export const AssetActions = ({ isLoaded, assetId, accountId }: AssetActionProps)
     isConnected ? send.open({ asset: asset, accountId }) : handleWalletModalOpen()
   const handleReceiveClick = () =>
     isConnected ? receive.open({ asset: asset, accountId }) : handleWalletModalOpen()
+  const notEnoughCryptoToSell = (): boolean => bnOrZero(cryptoBalance).eq(0)
```

Comment:
This shouldn't be a function, react components inherently react (😛) on props changes so if the concern is that it wouldn't react on it if a variable is used, no worries!

Also regarding good practices on naming and this specific terminology:
- checks evaluating to a bool are usually prefixed with `is`/`has`
- the "sell" terminology is wrong, this is about sending which is different

What about:

```javascript
const hasValidBalance = bnOrZero(cryptoBalance).gte(0)
```

And then inverting the happy and unhappy paths in l.45-47 to have the send copy on the left side and insufficient funds one on the right side of the ternary?

Reply:
Sure sounds good, will adjust in the `TradeInput` as well. But `isPositive()` won't work. It will return `true` when the value is `0`. I can alternatively call the `isZero` method and inverse that result.
<img width="236" alt="Screenshot 2022-02-01 at 09 40 49" src="https://user-images.githubusercontent.com/29582253/151929998-1b22feca-62ce-450a-98c8-6fcfbb2d8a85.png">

Reply:
but even with `isZero` it will return `false` for negative numbers, so there needs to be a comparison with 0. 🤔

Reply:
Ohhh nice one lol. So `gte(0)` it is then, you should never have a negative number in this case but my OCD will kick out knowing this could happen :D

Reply:
yup same, better be extra safe and also make sure the impossible cases don't break things :D